tests/driver_srf04/main.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test for srf04 ultra sonic range finder driver
+ *
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "srf04_params.h"
+#include "srf04.h"
+
+int main(void)
+{
+    puts("SRF04 range finder example");
+
+    srf04_t dev;
+    srf04_init(&dev, &srf04_params[0]);
+    int distance;
+
+    while (1) {
+        distance = srf04_get_distance(&dev);
+        if (distance < SRF04_OK) {
```

Comment:
This check is weird since the function can return a distance or error codes (negative values). Maybe just compare to 0 ? Or use a switch and address each potential error code ?

Reply:
Comparing to SRF04_OK is the same thing, since the struct defines it as zero and all errors would evaluate negative. I could distinguish between errors, but i deem it not necessary.

Reply:
I thought about your concern and now i believe to understand what you meant. Distance should only be SRF04_OK (zero), when srf04_init returns successfully. After the first get_distance (wich happens before) it will either have a valid distance (greater or equal to 0) or give an error. So comparing to SRF04_OK is just meant to be a clean way of checking for new data.
As the sensor only measures values from a minimum distance of i think 3 cm it should always be greater then zero when successfull.

Reply:
I'd say the check is fine. Maybe more intuitive would simply be to test for `distance < 0`, semantically considering the `0` as distance value instead of some arbitrarily defined named value. But I'd say this is up to the creative freedom of the author :-)